source/mesh_deformation/interface.cc

```diff
@@ -1060,14 +1060,33 @@ namespace aspect
             {
               AffineConstraints<double> user_level_constraints;
               user_level_constraints.reinit(relevant_dofs);
+              // if there is only one boundary using the tangential velocity,
+              // we won't have cycle in the constraints, so we can use the
+              // more efficient function: compute_no_normal_flux_constraints_shell.
```

Comment:
Why is it more efficient? Is it really worth it to keep both around? I would delete this one.